Source/APIClient/Request.swift

```diff
@@ -15,17 +15,35 @@ public enum HTTPMethod: String {
     case DELETE
 }
 
+public enum MediaType: String {
+    case json = "application/json"
+    case javascript = "application/javascript"
+    case xml = "application/xml"
+    case zip = "application/zip"
+    case urlEncodedForm = "application/x-www-form-urlencoded"
+    case multipartForm = "multipart/form-data"
+    case png = "image/png"
+    case jpeg = "image/jpeg"
+    case gif = "image/gif"
+    case mpeg = "audio/mpeg"
+    case vorbis = "audio/vorbis"
+    case css = "text/css"
+    case html = "text/html"
+    case plainText = "text/plain"
+}
+
 public protocol Request {
     static var baseURL: String { get }
     static var acceptHeader: String? { get }
+    static var credentialProvider: CredentialProviding? { get }
```

Comment:
I guess I changed this to `static`, but I don't remember if I ran this by you or not... Is this a good thing?